pkg/ccl/changefeedccl/alter_changefeed_stmt.go

```diff
@@ -682,7 +682,17 @@ func generateAndValidateNewTargets(
 		hasSelectPrivOnAllTables = hasSelectPrivOnAllTables && hasSelect
 		hasChangefeedPrivOnAllTables = hasChangefeedPrivOnAllTables && hasChangefeed
 	}
-	if err := authorizeUserToCreateChangefeed(ctx, p, sinkURI, hasSelectPrivOnAllTables, hasChangefeedPrivOnAllTables); err != nil {
+	if len(prevDetails.TargetSpecifications) == 0 {
+		return nil, nil, hlc.Timestamp{}, nil, errors.AssertionFailedf("target specifications should not be empty")
+	}
+	changefeedLevel := func() tree.ChangefeedLevel {
+		if prevDetails.TargetSpecifications[0].Type == jobspb.ChangefeedTargetSpecification_DATABASE {
```

Comment:
Is this even possible? I thought we restricted ALTER CHANGEFEED to not work for DB-level changefeeds